tests/entrypoints/test_openai_server.py

```diff
@@ -8,9 +8,69 @@
 import ray  # using Ray for overall ease of process management, parallel requests, and debugging.
 import openai  # use the official client for correctness check
 
+# imports for guided decoding tests
+import json
+import jsonschema
+import re
+
 MAX_SERVER_START_WAIT_S = 600  # wait for server to start for 60 seconds
 MODEL_NAME = "HuggingFaceH4/zephyr-7b-beta"  # any model with a chat template should work here
 
+TEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string"
+        },
+        "age": {
+            "type": "integer"
+        },
+        "skills": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "maxLength": 10
+            },
+            "minItems": 3
+        },
+        "work history": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "company": {
+                        "type": "string"
+                    },
+                    "duration": {
+                        "type": "string"
+                    },
+                    "position": {
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "company",
+                    "position"
+                ]
+            }
+        }
+    },
+    "required": [
+        "name",
+        "age",
+        "skills",
+        "work history"
+    ]
+}
+
+# NOTE: outlines' underlying regex library (interegular) doesn't support
+# ^ or $ or \b, kinda annoying
```

Comment:
See second point in [this section of the greenery documentation 
](https://github.com/qntm/greenery?tab=readme-ov-file#unsupported-constructs). `interegular` is a fork of greenery.

Reply:
I see, thanks for clarifying this!